ext/LogDensityProblemsADADTypesExt.jl

```diff
@@ -24,8 +24,10 @@ The AD configuration specified by `ad` is forwarded to the corresponding calls o
 """
 LogDensityProblemsAD.ADgradient(::ADTypes.AbstractADType, ℓ)
 
-function LogDensityProblemsAD.ADgradient(::ADTypes.AutoEnzyme, ℓ)
-    return LogDensityProblemsAD.ADgradient(Val(:Enzyme), ℓ)
+function LogDensityProblemsAD.ADgradient(ad::ADTypes.AutoEnzyme, ℓ)
+    # Default to Reverse mode but can't specifically say Enzyme
+    isnothing(ad.mode) && return LogDensityProblemsAD.ADgradient(Val(:Enzyme), ℓ)
+    return LogDensityProblemsAD.ADgradient(Val(:Enzyme), ℓ; mode=ad.mode)
```

Comment:
I was confused when reading this, maybe we should just allow `mode::Nothing` in `ADgradient(Val(:Enzyme), ...)` and use a `Enzyme.ReverseMode` default there?
```suggestion
    return LogDensityProblemsAD.ADgradient(Val(:Enzyme), ℓ; mode=ad.mode)
```

Reply:
I can move the check to ADgradient, but there will always need to be a check right? Like in ADgradient(Val(:Enzyme), args...) I'll need to check if mode is nothing and is so do the same replacement.

Reply:
I guess I went for this approach since the function below utilized a similar check. Again I am happy to move the check to ADgradient if that's preferred!

Reply:
> I guess I went for this approach since the function below utilized a similar check.

Good point, I hadn't seen the function below. Could we use the same style as below?
```suggestion
    if ad.mode === nothing
        # Use default mode (Enzyme.ReverseMode)
        return LogDensityProblemsAD.ADgradient(Val(:Enzyme), ℓ)
    else
        return LogDensityProblemsAD.ADgradient(Val(:Enzyme), ℓ; mode=ad.mode)
    end
```